tests/functional_test_base.go

```diff
@@ -119,7 +121,13 @@ func (s *FunctionalTestBase) setupSuite(defaultClusterConfigFile string, options
 
 	clusterConfig, err := GetTestClusterConfig(defaultClusterConfigFile)
 	s.Require().NoError(err)
-	clusterConfig.DynamicConfigOverrides = s.dynamicConfigOverrides
+	clusterConfig.DynamicConfigOverrides = util.CloneMapNonNil(s.dynamicConfigOverrides)
+	maps.Copy(clusterConfig.DynamicConfigOverrides, map[dynamicconfig.Key]any{
+		dynamicconfig.HistoryScannerEnabled:    false,
+		dynamicconfig.TaskQueueScannerEnabled:  false,
+		dynamicconfig.ExecutionsScannerEnabled: false,
+		dynamicconfig.BuildIdScavengerEnabled:  false,
```

Comment:
oh wait, is this going to override the suite-specific overrides? advanced visibility test needs build id scavenger. so I think the copy needs to go in the opposite order

Reply:
Switched the order